src/stdune/pp.ml

```diff
@@ -21,7 +21,7 @@ let rec map_tags t ~f =
   match t with
   | Nop -> Nop
   | Seq (a, b) -> Seq (map_tags a ~f, map_tags b ~f)
-  | Concat (sep, l) -> Concat (map_tags sep ~f, List.map l ~f:(map_tags ~f))
+  | Concat (sep, l) -> Concat (map_tags sep ~f, List_map.map l ~f:(map_tags ~f))
```

Comment:
`map_tags` is still not recursive by the way. How do we know this is enough to fix the problem?

Reply:
I used the test cases from the linked issues. I made blackbox tests, but they trigger the error only when the diff is printed, so a test case would need to be several megabytes of text, which is why I didn't include it. Thinking again, I can probably just silence the dune output.

Reply:
You could also just generate the file as part of the test. Hopefully the test isn't too slow though. The tests are already not very speedy.

Reply:
I did that; the test takes about 3 seconds. This can be sped up a bit using `ulimit` to decrease the maximum stack size but that's probably not very portable.